sdks/java/core/src/main/java/org/apache/beam/sdk/util/ReduceFnRunner.java

```diff
@@ -580,14 +581,16 @@ public void onTimer(TimerData timer) throws Exception {
           "ReduceFnRunner.onTimer: Note that timer {} is for non-ACTIVE window {}", timer, window);
     }
 
-    // If this is an end-of-window timer then, we need to set a GC timer
+    // If this is an end-of-window timer then we'll need to set a garbage collection timer.
     boolean isEndOfWindow = TimeDomain.EVENT_TIME == timer.getDomain()
-      && timer.getTimestamp().equals(window.maxTimestamp());
+                            && timer.getTimestamp().equals(window.maxTimestamp());
 
     // If this is a garbage collection timer then we should trigger and garbage collect the window.
-    Instant cleanupTime = window.maxTimestamp().plus(windowingStrategy.getAllowedLateness());
-    boolean isGarbageCollection =
-        TimeDomain.EVENT_TIME == timer.getDomain() && timer.getTimestamp().equals(cleanupTime);
+    // Be as generous as possible with the test to protect against accidental overflow/clipping
```

Comment:
Agreed and I'd take it further: Any time past the garbage collection time, we may GC. We don't actually have to care about why we received a wake-up. It could be a timer we set or any other impetus. The less we discuss exactly what, the more robust this is going to be.